Mathlib/NumberTheory/LSeries/AbstractFuncEq.lean

```diff
@@ -0,0 +1,450 @@
+/-
+Copyright (c) 2024 David Loeffler. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Loeffler
+-/
+
+import Mathlib.Analysis.MellinTransform
+
+/-!
+# Abstract functional equations for Mellin transforms
+
+This file formalises a general version of an argument used to prove functional equations for
+zeta and L functions.
+
+### FE-pairs
+
+We define a *weak FE-pair* to be a pair of functions `f, g` on the positive reals which are
+locally integrable, have the form "constant" + "rapidly decaying term" at `∞`, and satisfy a
+functional equation of the form
+
+`f (1 / x) = ε * x ^ k * g x`
+
+for some constants `k ∈ ℝ` and `ε ∈ ℂ`. (Modular forms give rise to natural examples
+with `k` being the weight and `ε` the global root number; hence the notation.) We could arrange
+`ε = 1` by scaling `g`; but this is inconvenient in applications so we set things up more generally.
+
+A *strong FE-pair* is a weak FE-pair where the constant terms of `f` and `g` at `∞` are both 0.
+
+The main property of these pairs is the following: if `f`, `g` are a weak FE-pair, then the Mellin
+transforms `Λ` and `Λ'` of `f - f₀` and `g - g₀` respectively both have meromorphic continuation
```

Comment:
Do you want to say what $f_0$ and $g_0$ are here? I agree it is clear from the code, but in this (very nice!) introduction they might also appear and I cannot find them.